spring-core/src/main/java/org/springframework/core/KotlinReflectionParameterNameDiscoverer.java

```diff
@@ -75,11 +75,16 @@
 	private String[] getParameterNames(List<KParameter> parameters) {
 		List<KParameter> filteredParameters = parameters
 				.stream()
-				.filter(p -> KParameter.Kind.VALUE.equals(p.getKind()))
+				// SPR-16119: Extension receivers of extension methods must be included as they appear as normal method
+				//            parameters in Java
+				.filter(p -> KParameter.Kind.VALUE.equals(p.getKind()) || KParameter.Kind.EXTENSION_RECEIVER.equals(p.getKind()))
 				.collect(Collectors.toList());
 		String[] parameterNames = new String[filteredParameters.size()];
 		for (int i = 0; i < filteredParameters.size(); i++) {
-			String name = filteredParameters.get(i).getName();
+			KParameter parameter = filteredParameters.get(i);
+			// extension receivers are not explicitly named, but require a name for Java interoperability
+			// $receiver is not a valid Kotlin identifier, but valid in Java, so it can be used here
+			String name = KParameter.Kind.EXTENSION_RECEIVER.equals(parameter.getKind())  ? "$receiver" : parameter.getName();
```

Comment:
As described in my JIRA issue comment, I think we should get the actual parameter name from `Method` instead of introducing `$receiver`.

Reply:
Except that there is no actual parameter name. We either will always get `arg0` (which is pretty useless) or `$receiver` if Kotlin's `javaParameters` flag is set.

Reply:
Good point, let's keep the meaningful parameter name.